fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java

```diff
@@ -838,19 +838,26 @@ public RemoteFileInfoSource getRemoteFilesAsync(Table table, GetRemoteFilesParam
         }
 
         IcebergTable icebergTable = (IcebergTable) table;
-        List<ScalarOperator> scalarOperators = Utils.extractConjuncts(params.getPredicate());
-        ScalarOperatorToIcebergExpr.IcebergContext icebergContext = new ScalarOperatorToIcebergExpr.IcebergContext(
-                icebergTable.getNativeTable().schema().asStruct());
-        Expression icebergPredicate = new ScalarOperatorToIcebergExpr().convert(scalarOperators, icebergContext);
-        RemoteFileInfoSource baseSource = buildRemoteInfoSource(icebergTable, icebergPredicate, snapshotId.get());
+
+        String dbName = table.getCatalogDBName();
+        String tableName = table.getCatalogTableName();
+        PredicateSearchKey predicateSearchKey = PredicateSearchKey.of(dbName, tableName, snapshotId.get(), param.getPredicate());
+        RemoteFileInfoSource baseSource;
+        if (splitTasks.containsKey(predicateSearchKey)) {
```

Comment:
RemoteFileInfoSource should be consumed at once,  so if you cache it,  the next time you use it, it will never emit any files.